source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -156,6 +156,39 @@ Filters are useful when you need an immediate response that directly modifies th
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Disadvantages**                                                  | - Can increase complexity in debugging and maintenance because it adds more processing layers.                | - Separation from the core makes long-term maintenance challenging.                                                                |
+|                                                                    | - Limited by existing events and filters available in the Open edX ecosystem, although new events and         | - High maintenance burden for upgrades and compatibility.                                                                          |
+|                                                                    |   filters can be proposed and added to the framework.                                                         | - Requires manual integration with the Open edX platform for each new release.                                                     |
+|                                                                    | - Might increase complexity in plugins for highly customized features.                                        | - Requires deep understanding of the platform to modify it correctly.                                                              |
+|                                                                    | - Although it is not recommended to do so, a plugin developer might introduce an edx-platform dependency,     |                                                                                                                                    |
+|                                                                    |   mainly when using filters, when there is no alternative for getting certain types of data.                  |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Community Compatibility**                                        | High, as the community encourages reusable implementations and the contribution of new extension points.      | Low, as changes in forks are usually highly customized not designed for others to reuse.                                           |
```

Comment:
> Low, as changes in forks are usually highly customized not designed for others to reuse.

And discoverability is low. Most probably no one will ever know "you did that".

Reply:
I added a note about discoverability here: https://github.com/openedx/docs.openedx.org/pull/617/commits/67bd6f7c4060421d7b2dea42c6affffa8a8d6acd#diff-01d0bcd5d9113aaa8a7f3fde94c7e5ca100265eebebbad867dbf4de21123d451R185